### ISSUE TITLE
speeding up step() and get_and_compare()

### ISSUE THREAD

**Issue by anandijain** [anandijain/sips#17]
tl;dr: **The current step is too slow, especially with high event counts.**

_I'll make sub issues for all of these_

TODO:

    1. Formally merge new_lines and collate.
    2. Remove bov_main.py 
    3. Separate pre and live game requests 
        * Rate limit pre-game requests

We want to be nice to the servers and abide by their request limits, but we also want to be fast. Currently, the async requests are fully disabled (I think), as the host would disconnect much more often, but this is definitely the ideal solution. If we had different IPs for the requesters, we could really get all the data in one go.

It would be nice to try to request/parse each unique site in tandem with the others. **This is the best path forward, when incorporating get_and_compare() from collate.py.** The data will be more accurate as the data for each site will be considered from the same time (we want to make this the as much the reality as possible). Another important add is to **only get live data, or at least only request espn if the bov data event status shows IN_PROGRESS.**

Current speed testing without collating espn (bov only):
The logs reflect that step time is **O(n)** where n is the number of events/games.
**Over ~5000 steps: the average time_diff / average num_events = 1.08 seconds**
This means that when we have 100 events, we can expect a step time of ~100 seconds, which is way too slow.

Horizon/Not viable:

    I have thought about trying to use reinforcement learning to handle some of the logistics parts of the main parser in the following ways:

    1. minimizing number of requests by penalizing requests where there is no new information and reward requests which have new data
    2. negative reward for host disconnect, zero reward for else
    3. combining above
